### DISCUSSION TITLE
Custom Queries "Function" What is it's purpose? What does it affect?

### DISCUSSION THREAD

**Discussion by JayP718** [Budibase/budibase#8451] [Help] [answered]
Hi,

I've used the function field within a custom query but I'm not sure what it effects and if it even matters. The docs here [custom-queries](https://docs.budibase.com/docs/custom-queries) also seem a bit sparse on that topic.

So far the only affect I've seen is that when the function is Read it doesn't show the auto notification when it's ran.

Thanks for anyone who can provide some pointers.

Best,

JP

**Comment by melohagan** [comment, marked as answer]
Hey @JayP718 

The function determines the CRUD operation, with some datasources including additional functions such as List, Activate, or anything else that might relate to the particular datasource.

A number of the datasources (such as the Custom SQL queries) simply execute a query regardless of which function is selected.
This is a hangover from when the interface required the CRUD functions to be provided even if they weren't needed. This has since been changed so that datasources can now be built with any number of relevant functions.

**Reply by JayP718** [reply]
Gotcha, thanks Mel.
